### PR TITLE
Use internal pull-up in pico_spi_sd_card example

### PR DESCRIPTION
**Pull request by ithinuel**
Depends on:
- https://github.com/rp-rs/rp-hal/pull/585
- https://github.com/rp-rs/ws2812-pio-rs/pull/25
- https://github.com/rp-rs/i2c-pio-rs/pull/23

### CODE REVIEW

File: boards/rp-pico/examples/pico_spi_sd_card.rs [hunk 1]
```diff
@@ -39,37 +39,6 @@
...
-//!  """"""""""""""""                                     |       |
-//!                                                       |       |
-//!                                                       .........
-//!                                                       |20   21|
-//!                                                        """""""
-//! Symbols:
-//!     - (+) crossing lines, not connected
-//!     - (o) connected lines
-//! ```
-//!
```
Comment (jannic) [inline boards/rp-pico/examples/pico_spi_sd_card.rs:72]: Why do you remove the schematic?

Comment (jannic) [reply boards/rp-pico/examples/pico_spi_sd_card.rs:72]: Ok, that makes sense. I like examples to describe the necessary cabling, but without the resistors, there's not much left in the diagram.

Also, I think the relationship between pull-ups and data rate hold for I2C, but not for SPI, which is actively driven. One more reason to remove this comment.

**Comment by ithinuel** [comment]
Indeed, but it seems like @jannic already integrated those changes in: https://github.com/rp-rs/rp-hal-boards/commit/bcbe172d6a13828cc82d8ce3736e1f09125701ed

**Review by jannic — APPROVED** [review]
:+1:

**Pull request merged**